apis/k8ssandra/v1alpha1/constants.go

```diff
@@ -20,6 +20,12 @@ const (
 	// are still deployed serially).
 	DcReplicationAnnotation = "k8ssandra.io/dc-replication"
 
+	// RebuildSourceDcAnnotation tells the operation the DC from which to stream when
+	// rebuilding a DC. If not set the operator will choose the first DC. The value for
+	// this annotation must specify the name of a CassandraDatacenter whose Ready
+	// condition is true.
+	RebuildSourceDcAnnotation = "k8ssandra.io/rebuild-src-dc"
```

Comment:
A quick question, we leave this (and the other annotation indicating replicationFactors) to the K8ssandraCluster. Should we remove it after the processing has finished to ensure these rules are reset if someone wants to add another?

Reply:
Good question. I have asked myself the same. Part of the reason I didn't implement behavior to remove them is that I thought users might want to see that state for auditing purposes and what not. Now I also think it could be good to remove them because the user might forget about them. 

In the case of the `k8ssandra.io/dc-replication`, it won't matter. It only applies to the dc being added. So if it left in place and another DC is added and the annotation is ignored it should be the same as not specifying the annotation.

For the `k8ssandra.io/rebuild-src-dc` annotation, I don't think it's a big deal to leave it. Doing so would basically only alter the default behavior.